File: implement-shell-tools/cat/cat.js
const fs = require("fs");

const args = process.argv.slice(2);

let flagN = false;
let flagB = false;
const files = [];

for (const arg of args) {
  if (arg === "-n") flagN = true;
  else if (arg === "-b") flagB = true;
  else files.push(arg);
}

let lineNumber = 1;

for (const file of files) {
  const content = fs.readFileSync(file, "utf-8");
  const lines = content.split("\n");

  // readFileSync agrega un \n final que genera una línea vacía extra
  if (lines[lines.length - 1] === "") lines.pop();

  for (const line of lines) {
    if (flagB) {
      if (line === "") {
        console.log("");
      } else {
        console.log(`     ${lineNumber}\t${line}`);
        lineNumber++;
      }
    } else if (flagN) {
      console.log(`     ${lineNumber}\t${line}`);
      lineNumber++;
    } else {
      console.log(line);
    }
  }
}

File: implement-shell-tools/cat/cat.py
import sys
import os

args = sys.argv[1:]

flag_n = False
flag_b = False
files = []

for arg in args:
    if arg == "-n":
        flag_n = True
    elif arg == "-b":
        flag_b = True
    else:
        files.append(arg)

line_number = 1

for file in files:
    with open(file, "r") as f:
        lines = f.readlines()
    for line in lines:
        line = line.rstrip("\n")
        if flag_b:
            if line == "":
                print("")
            else:
                print(f"     {line_number}\t{line}")
                line_number += 1
        elif flag_n:
            print(f"     {line_number}\t{line}")
            line_number += 1
        else:
            print(line)